### PR TITLE
apply token if exist

### PR DESCRIPTION
**Pull request by javasoze**

### CODE REVIEW

**Comment by javasoze** [comment]
tested locally against a local deployment against api

File: utils/rpc.py [hunk 1]
```diff
@@ -4,9 +4,24 @@
 import urllib3
 from urllib3.util import Retry
 import time
+import os
 
 
 class RPC(object):
+    @staticmethod
+    def readToken(tok_file=None):
```
Comment (keitaf) [inline utils/rpc.py]: can we just set the default value here?
```
def readToken(tok_file='~/.dashbase/dashbase.token')
```

File: utils/rpc.py [hunk 2]
```diff
@@ -19,6 +34,10 @@ def __init__(self, host_port, num_tries=5, delay_in_sec=5):
         self.conn = None
         self.init_connection()
 
+    def set_token(self, token):
+        if token is not None:
```
Comment (keitaf) [inline utils/rpc.py]: nit: `if token:`

File: utils/rpc.py [hunk 3]
```diff
@@ -4,9 +4,24 @@
...
+        try:
+            f = open(tok_file, 'r')
+            tok = f.readline()
+            f.close()
+            return tok
+        except Exception:
+            print ("no token found at: " + tok_file)
+            return None
+
     def __init__(self, host_port, num_tries=5, delay_in_sec=5):
```
Comment (keitaf) [inline utils/rpc.py]: can we load and set token in the constructor??
```
def __init__(..., tok_file='~/.dashbase/dashbase.token'):
  load and set token here
```

File: utils/rpc.py (hunk 1 above)
Comment (javasoze) [reply utils/rpc.py]: done

File: utils/rpc.py (hunk 2 above)
Comment (javasoze) [reply utils/rpc.py]: done

File: utils/rpc.py (hunk 3 above)
Comment (javasoze) [reply utils/rpc.py]: that is a much better idea and makes the diff smaller and cleaner overall! Thanks!

**Pull request merged**